scripts/portfolioView.js

```diff
@@ -0,0 +1,41 @@
+(function(module) {
+
+  var projectView = {};
+
+  projectView.populateFilters = function() {
+    $('article').each(function() {
+      if (!$(this).hasClass('template')) {
+        val = $(this).attr('data-category');
+        var optionTag = '<option value="' + val + '">' + val + '</option>';
+        if ($('#category-filter option[value="' + val + '"]').length === 0) {
+          $('#category-filter').append(optionTag);
+        }
+      }
+    });
+  };
+
+  projectView.categoryFilter = function() {
+    $('#category-filter').on('change', function() {
+      if ($(this).val()) {
+        $('article').hide();
+        $('article[data-category="' + $(this).val() + '"]').fadeIn();
+      } else {
+        $('article').fadeIn();
+      }
+    });
+  };
+
+  projectView.initIndexPage = function() {
+    if ($('#projects article').length === 0 ) {
```

Comment:
We talked about this a little on Monday.  This will keep you from continually appending more articles each time.  The problem I thought of after we talked about it though occurs if the data is updated.  So if the articles you got back were updated articles and you wanted to replace the current ones, then it would not do that.  A better solution might be to remove the existing articles every time and append the new ones.